sphinx_immaterial/cpp_domain_fixes.py

```diff
@@ -192,8 +201,15 @@ def describe_signature(
 This allows additional customizations of those functions to access the symbol.
 """
 
+DOMAIN_CLS_TYPE = Union[sphinx.domains.c.CDomain, sphinx.domains.cpp.CPPDomain]
+SYMBOL_CLS_TYPE = Union[sphinx.domains.c.Symbol, sphinx.domains.cpp.Symbol]
```

Comment:
This should be `Union[Type[sphinx.domains.c.Symbol], Type[sphinx.domains.cpp.Symbol]]`

Reply:
To be more precise, in some uses you need the `Type` version and in other uses you need the version as you have it here.

e.g. the `self` parameter types should be just `Union[sphinx.domains.c.Symbol, sphinx.domains.cpp.Symbol]`, but when monkey patching we are usually modifying the type object itself.